pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -110,69 +52,177 @@ def check_parser_compatibility(cls, file):
-        lookup = {}
-        for pep in self.root.findall(".//{*}Peptide"):
-            id = pep.attrib.get("id", "")
-            lookup[id] = {"modifications": []}
-            lookup[id]["sequence"] = pep.find(".//{*}PeptideSequence").text
-            for child in pep.findall(".//{*}Modification"):
-                lookup[id]["modifications"].append(
-                    f"{modification_mass_map[child.attrib['monoisotopicMassDelta']]}:{child.attrib['location']}"
+        Returns:
+            fixed_mods (dict): residues and corresponding names of fixed modifications
+            mod_mass_map (dict): mapping of masses to modification names
+        """
+        fixed_mods = {}
+        mod_mass_map = {}
+
+        mod_name = ""
+        modification_information = False
+
+        for event, entry in etree.iterparse(self.input_file):
+            entry_tag = entry.tag
+
+            if entry_tag.endswith("AdditionalSearchParams"):
+                modification_information = True
+            elif modification_information is True:
+                if entry_tag.endswith("cvParam"):
+                    mod_name = entry.attrib["name"]
+                elif entry_tag.endswith("SearchModification"):
+                    mod_mass_map[entry.attrib["massDelta"]] = mod_name
+                    if entry.attrib["fixedMod"] == "true":
+                        fixed_mods.update({entry.attrib["residues"]: mod_name})
+                elif entry_tag.endswith("ModificationParams"):
+                    return fixed_mods, mod_mass_map
+            entry.clear()
+
+    def get_peptide_lookup(self):
+        """Retrieve peptide ids with their corresponding sequences and modifications from xml.
+
+        Returns:
+            peptide_lookup (dict): peptide id with corresponding modifications and sequence
+        """
+        peptide_lookup = {}
+
+        modifications = ""
+        sequence = {}
+        peptide_information = False
+
+        for event, entry in etree.iterparse(self.input_file):
+            entry_tag = entry.tag
+
+            if entry_tag.endswith("DBSequence"):
+                peptide_information = True
+            elif peptide_information is True:
+                if entry_tag.endswith("PeptideSequence"):
+                    sequence = entry.text
+                    if len(self.fixed_mods) > 0:
+                        sequence_mod_map = self.map_mods_sequences(sequence)
+                        modifications = sequence_mod_map + ";"
+                    else:
+                        modifications = ""
+                elif entry_tag.endswith("Modification"):
+                    mass = entry.attrib["monoisotopicMassDelta"]
+                    location = entry.attrib["location"]
+                    mass_name = self.mod_mass_map[mass]
+                    modifications += mass_name + ":" + location + ";"
+                elif entry_tag.endswith("Peptide"):
+                    modifications = modifications.rstrip(";").lstrip(";")
```

Comment:
if modifications is a list and one uses ";".join(modifications) at the end (i.e. l.137), the rstrip, lstrip, + ";" elements can be spared (readability 💯 )